general/backends.hpp

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2010-2020, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#ifndef MFEM_BACKENDS_HPP
+#define MFEM_BACKENDS_HPP
+
+#include "../config/config.hpp"
+
+#ifdef MFEM_USE_CUDA
+#include <cuda_runtime.h>
+#include <cuda.h>
+#endif
+#include "cuda.hpp"
+
+#ifdef MFEM_USE_HIP
+#include <hip/hip_runtime.h>
+#endif
+#include "hip.hpp"
```

Comment:
I would expect `cuda.hpp` and `hip.hpp` to not be included when Cuda and Hip are not used respectively, is it necessary to include them?

Reply:
The Cuda/Hip API (`CuMemAlloc`) is made available through these two files for now.
It's been used in the `MemoryManager`.
We could be able to hide it - that's something we can think of while refactoring.